libsolutil/StringUtils.h

```diff
@@ -101,109 +101,117 @@ std::string joinHumanReadablePrefixed
 		return _separator + joinHumanReadable(_list, _separator, _lastSeparator);
 }
 
-/// Same as @ref formatNumberReadable but only for unsigned numbers
+/// Formats large numbers to be easily readable by humans.
+/// Returns decimal representation for smaller numbers; hex for large numbers.
+/// "Special" numbers, powers-of-two and powers-of-two minus 1, are returned in
+/// formulaic form like 0x01 * 2**24 - 1.
+/// @a T can be any integer variable, will typically be u160, u256 or bigint.
+/// @param _value to be formatted
+/// @param _useTruncation if true, internal truncation is also applied,
+/// like  0x5555...{+56 more}...5555
+/// @example formatNumberReadable((u256)0x7ffffff) = "0x08 * 2**24"
+/// @example formatNumberReadable(-57896044618658097711785492504343953926634992332820282019728792003956564819968) = -2**255
 template <class T>
-inline std::string formatUnsignedNumberReadable (
+inline std::string formatNumberReadable(
 	T const& _value,
 	bool _useTruncation = false
 )
 {
 	static_assert(
-		std::is_same<bigint, T>::value || !std::numeric_limits<T>::is_signed,
-		"only unsigned types or bigint supported"
-	); //bigint does not carry sign bit on shift
+		std::numeric_limits<T>::is_integer,
+		"only integer numbers are supported"
+	);
+
+	bool isNegative = _value < 0;
+	bigint temp;
+	bigint absoluteValue = isNegative ? (bigint(-1) * _value) : bigint(_value);
```

Comment:
Maybe a little nitpick, but it seems to me that `absoluteValue` means how far from zero the number is and then it cannot be negative. What about `decimalValue` (or `signedValue` as suggested before)?

Reply:
I think signedValue would be better